packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Expressions/ValueExpressions/BinaryDataExpression.cs

```diff
@@ -36,5 +36,8 @@ public static BinaryDataExpression FromObjectAsJson(ValueExpression data)
 
         public static BinaryDataExpression FromString(ValueExpression data)
             => new(InvokeStatic(nameof(BinaryData.FromString), data));
+
+        static BinaryDataExpression ITypedValueExpressionFactory<BinaryDataExpression>.Create(ValueExpression untyped)
```

Comment:
Is there any reason we don't add explicit cast operators here instead?

Also should this fail if the ValueExpression given is not a BinaryDataExpression?  How would we check that?

Reply:
this is supposed to wrap a untyped expression to this specific typed expression, works as the same as its ctor.
I use this to work as a constraint of "it has a ctor with parameter of valueexpression"

Reply:
If this is the case "it has a ctor with parameter of valueexpression", why not just let the user call new BoolExpression(valueExpression) vs BoolExpression.Create(valueExpression).  in dotnet land the new ctor syntax is much more discoverable?

Reply:
the whole point is not for how our use would construct this, despite this may have affects on that part.
This is trying to introduce a "generic constraint" so that we could have:
```
MethodBodyStatement Declare<T>(string name, T expression, out T variable)
```
as I have written in the description.
Adding this does not remove the ctor though.